### PR TITLE
KOGITO-6540 : Stunner - Ability to set the priority of a task as a process variable in the BC BPMN editor

### PR DESCRIPTION
**Pull request by inodeman**
Hi @romartin this is the PR for the Task Priority Change, ready for review

### CODE REVIEW

File: packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/fromstunner/tasks/TaskConverter.java [hunk 1]
```diff
@@ -167,7 +168,7 @@ private PropertyWriter userTask(Node<View<BaseUserTask>, ?> n) {
         p.setGroupId(executionSet.getGroupid().getValue());
         p.setSubject(executionSet.getSubject().getValue());
         p.setDescription(executionSet.getDescription().getValue());
-        p.setPriority(executionSet.getPriority().getValue());
+        p.setPriority(StringUtils.replaceIllegalCharsAttribute(executionSet.getPriority().getValue()));
```
Comment (romartin) [inline packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/fromstunner/tasks/TaskConverter.java:171]: Hey @inodeman 
Think that different from BC (kie-wb-common), this marshaller implementation is running on the client side. This way, I am not sure if `StringUtils.replaceIllegalCharsAttribute` is emulated properly here, also same thing for `StringUtils.revertIllegalCharsAttribute`, which is being used on the class below.
Can you please veriify the resutls for those methods and check they're doing what's supposed to do? Oherwise, we'll have to find another client side library that does this.
Thanks!

**Review by romartin — APPROVED** [review]
Sorry for the confusing, the StringUtils class emulates properly the logic, as it's inside stunner core.
Approving.
Thanks!

File: packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/resources/org/kie/workbench/common/stunner/bpmn/resources/i18n/StunnerBPMNConstants.properties [hunk 2]
```diff
@@ -172,6 +172,13 @@ org.kie.workbench.common.stunner.bpmn.definition.property.connectors.ConditionEx
 org.kie.workbench.common.stunner.bpmn.definition.property.connectors.Priority.label=Priority
 org.kie.workbench.common.stunner.bpmn.definition.property.connectors.Priority.description=Priority
 
+#######################
+# TaskPriority
+org.kie.workbench.common.stunner.bpmn.definition.property.task.TaskPriority.label=Task Priority
+org.kie.workbench.common.stunner.bpmn.definition.property.task.TaskPriority.description=Sets the Priority of the Task
+org.kie.workbench.common.stunner.bpmn.definition.property.task.TaskPriority.helpMessage=Sets the MVEL #{expression} value or integer
```
Comment (LuboTerifaj) [inline packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/resources/org/kie/workbench/common/stunner/bpmn/resources/i18n/StunnerBPMNConstants.properties]: Could be this reformulated a little bit? I am wondering if "Sets" here does make sense.

Would it be also possible to add more complex message here similarly to tooltips in Multiple Instance sub-process for example?

**Review by LuboTerifaj — COMMENTED** [review]
Hello @inodeman ,

there is no any validation for allowed values in the Priority property, but as already mentioned in 
[kie-wb-common-PR3723](https://github.com/kiegroup/kie-wb-common/pull/3723), it is out of scope of this implementation.

Please have a look at my inline comment.

Otherwise, everything works as expected.

Thank you!

**Comment by romartin** [comment]
Hey @LuboTerifaj 
Yes good catch, the help message could be improved... so I jus ttried to rewrite it a bit more clear for end users. Let's see if you feel more confortable now.
Anyway please, this is SO SO URGENT, so don't spend much time here now.. we can change the message even in a commit later,, but cusotmer really needs this new feature.
Also please remember the PR for BC -> https://github.com/kiegroup/kie-wb-common/pull/3723
Thanks!

**Review by LuboTerifaj — APPROVED** [review]
It looks good to me now. Thank you!

**Comment by LuboTerifaj** [comment]
I have reported the issue about tooltip message here: https://issues.redhat.com/browse/KOGITO-6832

**Pull request merged**